conda/plugins/environment_exporters/yaml.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""Built-in conda YAML environment exporter plugin."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ...common.serialize import yaml_safe_dump
+from ...exceptions import CondaValueError
+from ..hookspec import hookimpl
+from ..types import CondaEnvironmentExporter
+
+if TYPE_CHECKING:
+    from ...models.environment import Environment
+
+
+def export_yaml(env: Environment) -> str:
+    """Export Environment to YAML format."""
+    # Use the model's own method that follows EnvironmentYaml.to_dict() pattern
+    env_dict = env.to_yaml_dict()
+    yaml_content = yaml_safe_dump(env_dict)
+    if yaml_content is None:
+        raise CondaValueError("Failed to export environment to YAML")
+    return yaml_content
+
+
+@hookimpl(tryfirst=True)  # Ensure built-in YAML exporter loads first and has priority
```

Comment:
It looks like the cli layer has a mechanism for selecting the default exporter implementation, and the the plugin types have a mechanism to ensure that all plugin names are unique, is this `tryfirst` needed?